Xamarin.Forms.Platform.Android/Renderers/StepperRenderer.cs

```diff
@@ -36,74 +40,33 @@ protected override void OnElementChanged(ElementChangedEventArgs<Stepper> e)
 
 			if (e.OldElement == null)
 			{
-				_downButton = new AButton(Context) { Text = "-", Gravity = GravityFlags.Center, Tag = this };
-				_downButton.SetHeight((int)Context.ToPixels(10.0));
-
-				_downButton.SetOnClickListener(StepperListener.Instance);
-
-				_upButton = new AButton(Context) { Text = "+", Tag = this };
-
-				_upButton.SetOnClickListener(StepperListener.Instance);
-				_upButton.SetHeight((int)Context.ToPixels(10.0));
-
 				var layout = CreateNativeControl();
-
-				layout.AddView(_downButton);
-				layout.AddView(_upButton);
-
+				StepperRendererManager.CreateStepperButtons(this, out _downButton, out _upButton);
+				layout.AddView(_downButton, new LinearLayout.LayoutParams(LayoutParams.WrapContent, LayoutParams.MatchParent));
+				layout.AddView(_upButton, new LinearLayout.LayoutParams(LayoutParams.WrapContent, LayoutParams.MatchParent));
 				SetNativeControl(layout);
 			}
 
-			UpdateButtonEnabled();
+			StepperRendererManager.UpdateButtons(this, _downButton, _upButton);
 		}
 
 		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
 		{
 			base.OnElementPropertyChanged(sender, e);
-
-			switch (e.PropertyName)
-			{
-				case "Minimum":
-					UpdateButtonEnabled();
-					break;
-				case "Maximum":
-					UpdateButtonEnabled();
-					break;
-				case "Value":
-					UpdateButtonEnabled();
-					break;
-				case "IsEnabled":
-					UpdateButtonEnabled();
-					break;
-			}
+			StepperRendererManager.UpdateButtons(this, _downButton, _upButton, e);
 		}
 
-		void UpdateButtonEnabled()
-		{
-			Stepper view = Element;
-			_upButton.Enabled = view.IsEnabled ? view.Value < view.Maximum : view.IsEnabled;
-			_downButton.Enabled = view.IsEnabled ? view.Value > view.Minimum : view.IsEnabled;
-		}
+		Stepper IStepperRenderer.Element => Element;
 
-		class StepperListener : Object, IOnClickListener
-		{
-			public static readonly StepperListener Instance = new StepperListener();
+		AButton IStepperRenderer.UpButton => _upButton;
 
-			public void OnClick(global::Android.Views.View v)
-			{
-				var renderer = v.Tag as StepperRenderer;
-				if (renderer == null)
-					return;
-
-				Stepper stepper = renderer.Element;
-				if (stepper == null)
-					return;
+		AButton IStepperRenderer.DownButton => _downButton;
 
-				if (v == renderer._upButton)
-					((IElementController)stepper).SetValueFromRenderer(Stepper.ValueProperty, stepper.Value + stepper.Increment);
-				else if (v == renderer._downButton)
-					((IElementController)stepper).SetValueFromRenderer(Stepper.ValueProperty, stepper.Value - stepper.Increment);
-			}
+		AButton IStepperRenderer.CreateButton()
+		{
+			var button = new AButton(Context);
+			button.SetHeight((int)Context.ToPixels(10.0));
```

Comment:
Where does this 10 come from?

Reply:
That was always there: https://github.com/xamarin/Xamarin.Forms/blob/0dd61b71ec33c566ddcdf0fcb3aa2ec177b67399/Xamarin.Forms.Platform.Android/Renderers/StepperRenderer.cs#L40

Reply:
Fair enough.